unity/CITools/BuildDriver/Program.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Runtime.InteropServices;
+using NiceIO;
+
+namespace BuildDriver;
+
+public class Program
+{
+    public static bool RunningOnYamato() => Environment.GetEnvironmentVariable("YAMATO_PROJECT_ID") != null;
+    public static async Task<int> Main(string[] args)
+    {
+        string architecture = GetArchitectures()[0]; // Get default arch for current system
+        string configuration = "Release";
+        bool silent = false;
+        bool skipBuild = false;
+        bool test = false;
+        bool zip = RunningOnYamato();
+        Task<NPath>? zipTask = null;
+        foreach (string arg in args)
+        {
+            if ((arg.StartsWith("--arch") || arg.StartsWith("--architecture="))
+                && !TryParseArgument(GetArchitectures(true), arg, out architecture))
+                return 1;
+            else if ((arg.StartsWith("--config=") || arg.StartsWith("--configuration="))
+                     && !TryParseArgument(new[] {"Release", "Debug"}, arg, out configuration))
+                return 1;
+            else if (arg.Equals("--silent") || arg.Equals("--s"))
+                silent = true;
+            else if (arg.Equals("--zip") || arg.Equals("--z"))
+                zip = true;
+            else if (arg.Equals("--skip-build"))
+                skipBuild = true;
+            else if (arg.Equals("--test"))
+            {
+                skipBuild = true; // Assume we've already built
+                test = true;
+            }
+        }
+
+        if (RunningOnYamato())
+        {
+            zipTask = SevenZip.DownloadAndUnzip7Zip();
+        }
+
+        Console.WriteLine("*****************************");
+        Console.WriteLine("Unity: Starting CoreCLR build");
+        Console.WriteLine($"\tPlatform:\t\t{RuntimeInformation.OSDescription}");
+        Console.WriteLine($"\tArchitecture:\t\t{architecture}");
+        Console.WriteLine($"\tConfiguration:\t\t{configuration}");
+        Console.WriteLine("*****************************");
+
+        GlobalConfig gConfig = new (){ Architecture = architecture, Configuration = configuration, Silent = silent};
+        // TODO: Do we need to build this every time?
```

Comment:
TODO

Reply:
This was actually something I wanted to ask you. Is there anything in your changes that would make it so we don't need to build the `managed.sln` every time? I thought you had mentioned that you had added something that would kick off the build for it.

Reply:
That doesn't ring any bells.  It needs to be built every time.  It's the copying of the dlls to the coreclr artifact directory that I added logic for.  But that logic won't run here because coreclr may not have ran yet.